trio/_util.py

```diff
@@ -135,6 +136,13 @@ def __init__(self, func, args, kwds):
                 if sys.exc_info()[1] is not value:
                     raise
 
+    def __enter__(self):
+        raise RuntimeError("use 'async with {func_name}(...)', not 'with {func_name}(...)'".format(func_name=self._func_name))
+
+    def __exit__(self):
+        assert False, """Never called, but should be defined""" # pragma: no-cover
```

Comment:
Should be `no cover`, not `no-cover`; it's not actually being recognized right now :-). Also better to put two spaces before the `#`, and to move it up one line, so:
```python
def __exit__(self):  # pragma: no cover
```